compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/Desugar.java

```diff
@@ -861,31 +860,39 @@ private BLangStatementExpression createIfElseFromConfigurable(BLangSimpleVariabl
     }
 
     private List<BLangVariable> desugarGlobalVariables(BLangPackage pkgNode, BLangBlockFunctionBody initFnBody) {
-        List<BLangVariable> globalVars = pkgNode.globalVars;
         List<BLangVariable> desugaredGlobalVarList = new ArrayList<>();
         SymbolEnv initFunctionEnv =
                 SymbolEnv.createFunctionEnv(pkgNode.initFunction, pkgNode.initFunction.symbol.scope, env);
 
-        globalVars.forEach(globalVar -> {
+        for (BLangVariable globalVar : pkgNode.globalVars) {
             this.env.enclPkg.topLevelNodes.remove(globalVar);
             // This will convert complex variables to simple variables.
             switch (globalVar.getKind()) {
                 case TUPLE_VARIABLE:
                     BLangNode blockStatementNode = rewrite(globalVar, initFunctionEnv);
                     List<BLangStatement> statements = ((BLangBlockStmt) blockStatementNode).stmts;
-                    for (int i = 0; i < statements.size(); i++) {
-                        BLangStatement bLangStatement = statements.get(i);
-                        // First statement is the virtual array created for the init expression.
-                        // Rest binding pattern array initialization will be desugared as a block hence add them
-                        // directly to the init function body.
-                        if (bLangStatement.getKind() == NodeKind.BLOCK || i == 0) {
-                            initFnBody.stmts.add(bLangStatement);
-                            continue;
-                        }
-                        BLangSimpleVariableDef simpleVarDef = (BLangSimpleVariableDef) bLangStatement;
-                        simpleVarDef.var.annAttachments = globalVar.getAnnotationAttachments();
-                        addToInitFunction(simpleVarDef.var, initFnBody);
-                        desugaredGlobalVarList.add(simpleVarDef.var);
+                    int statementSize = statements.size();
+                    if (statementSize == 0) {
+                        break;
```

Comment:
Or continue?

Do we need this check anyway? This can't be empty, right?

Reply:
Ah, missed that this was in the switch case. 

But do we need the check for 0?

Reply:
Yes, we can remove it. It creates a statement for RHS though the list BP is empty.